_posts/2023-03-09-solidity-developer-survey-2022-results.md

```diff
@@ -0,0 +1,451 @@
+
+The biggest pain point is “stack-too-deep”, with 33.6% of all votes, followed by missing memory optimizations (waste of memory) (24.4%) and redundant checks (e.g. in checked arithmetic) (11.8%). 9.9% say that compiler performance is their biggest issue.
+
+15.8% selected “other” and were able to specify their most significant pain point in a free text field. Most prominently mentioned: Contract size limit, error messages, and issues with debugging.
+
+![Solidity Pain Points](/img/2023/02/pain_points.png)
+
+### High-Impact Compiler Bugs
+
+As part of this year’s study, we were also curious to find out whether Solidity developers had been affected by any of the high-impact compiler bugs (codegen bugs that are announced with [Security Alerts](https://blog.soliditylang.org/category/security-alerts/) on the Solidity blog).
+
+Initially, 4.7% said yes. However, when asked which one they were affected by, only two out of the 63 people were able to point to actual Solidity vulnerabilities in the follow-up question. This gives room to assume the actual number of affected developers from this survey is significantly lower than 4.7%, and some respondents may have simply misunderstood the question. We will try to phrase this question more precisely in the next survey.
+
+![High Impact Bugs](/img/2023/02/security_bugs.png)
+
+## Language Design & Upcoming Features
+
+### Favorite Feature / Solidity Aspect
+
+Respondents most like Solidity's syntax, the simplicity with regards to learning, reading, coding and compiling and static typing.
+
+The most mentioned liked features in descending order were:
+
+* Syntax
+* Easy to... read/code/compile/learn
+* Simplicity
+* Static/strong/strict typing
+* Similarity to other languages (most mentioned JS/TS and object-orientedness, also mentioned: Rust, C++, Python)
+* Modifiers
+* Inline assembly
+* Mappings
+* Inheritance
+* (User-defined) types
+* SafeMath / checked and unchecked
+* Yul
+
+### Most Dreaded Aspect
+
+This year, we asked the survey audience a slightly different question: “If you could change one thing about Solidity, what would it be?”
+
+Most mentioned “change requests” in descending order were:
+
+* Fix stack-too-deep
+* Better array handling
+* Gas optimizations
+* Add decimal math (fixed point types / floating types)
+* Better error handling, descriptions
+* Better debugging
+* Higher contact bytecode size limit
+
+### Future Features
+
+#### Most Anticipated Feature
+
+Support for decimal numbers and generics was mentioned most often as the “most anticipated Solidity feature”.
+
+_⚠️ Similar to the previous year, we noted that respondents were using various different terms like "floats", "floating point arithmetic", "floating point number", "fixed point numbers", and "fixed point math". We categorized those as "decimal numbers"._
```

Comment:
Maybe "fractional numbers" would be more accurate? Floating-point and fixed-point numbers can be both decimal and binary.

Reply:
What is the term we use in our docs?

Reply:
It's all over the place:
- "fixed-point number"
- "decimal number"
- "fixed-point decimal number"
- "decimal fractional literals"
- "fractional rational number"
- "fractional values"

But an important bit of context is that it's in reference to what we already have in the language, i.e. the rational literals and the `ufixed`/`fixed` types. And those are fixed-point and decimal.

In the blog post we're referring a more general category so I'd rather say "fractional", but just to be clear, I think most people will understand "decimal" anyway so not a big deal. Just wanted to point out that it's IMO not 100% technically correct. And not having a consistent terminology for things in the language is a recurring issue for us.

Reply:
You're correct Kamil, we also referred to it as "fractional numbers" in the last survey results report. So will adjust it to that!